cmd/skopeo/flag.go

```diff
@@ -73,3 +73,37 @@ func (ob *optionalStringValue) String() string {
 	}
 	return ob.value
 }
+
+// optionalInt is a int with a separate presence flag.
+type optionalInt struct {
+	present bool
+	value   int
+}
+
+// optionalInt is a cli.Generic == flag.Value implementation equivalent to
+// the one underlying flag.Int, except that it records whether the flag has been set.
+// This is distinct from optionalInt to (pretend to) force callers to use
+// newoptionalInt
+type optionalIntValue optionalInt
+
+func newOptionalIntValue(p *optionalInt) cli.Generic {
+	p.present = false
+	return (*optionalIntValue)(p)
+}
+
+func (ob *optionalIntValue) Set(s string) error {
+	v, err := strconv.Atoi(s)
```

Comment:
`flag.Int` seems to use something else; if this promises to be equivalent, it must be equivalent.